### PR TITLE
newdevice: Add more sensor detection for Mikrotik devices #5387

### PR DESCRIPTION
**Pull request by geordish**
Added more sensors for Mikrotik. (#5387)

Could do with some testing, as my Mikrotik devices don't have these sensors!

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

### CODE REVIEW

File: includes/discovery/sensors/fanspeeds/mikrotik.inc.php [hunk 1]
```diff
@@ -1,28 +1,19 @@
...
-            $index            = $split_oid[(count($split_oid) - 1)];
-            $descr            = 'Fan '.$index;
-            $oid              = '.1.3.6.1.4.1.14988.1.1.3.9.'.$index;
-            $fanspeed         = (snmp_get($device, $oid, '-Oqv') / $divisor);
-            if ($fanspeed > 0) {
-                discover_sensor($valid['sensor'], 'fanspeed', $device, $oid, $index, $type, $descr, $divisor, '1', null, null, null, null, $fanspeed);
-            }
+        if (is_numeric($value)) {
+            $descr = $descr_prefix . ($index + 1);
+            discover_sensor($valid['sensor'], 'fanspeed', $device, $oid, $index, 'snmp', $descr, 1, 1, null, null, null, null, $value);
```
Comment (laf) [inline includes/discovery/sensors/fanspeeds/mikrotik.inc.php]: The change in oid here will cause old sensors to be deleted and new added. What's the reason for the change?

Comment (geordish) [inline includes/discovery/sensors/fanspeeds/mikrotik.inc.php]: The old OID that was being pulled doesn't return fan speed.
snmptranslate -On MIKROTIK-MIB::mtxrHlActiveFan
.1.3.6.1.4.1.14988.1.1.3.9

From the issue:
iso.3.6.1.4.1.14988.1.1.3.9.0 = STRING: "

On my Mikrotik:
.1.3.6.1.4.1.14988.1.1.3.9.0 = STRING: "n/a"

From the MIB (its a string):
mtxrHlActiveFan OBJECT-TYPE
    SYNTAX DisplayString
    MAX-ACCESS read-only
    STATUS current
    DESCRIPTION ""
    ::= { mtxrHealth 9 }

The 2 OIDs now polled are the correct ones.

I wonder if their MIB changed at some point.

Comment (laf) [inline includes/discovery/sensors/fanspeeds/mikrotik.inc.php]: Thanks for the clarification :)

Comment (Cheeze-It) [inline includes/discovery/sensors/fanspeeds/mikrotik.inc.php]: The old OID actually asks for which fan "stage" is actually being in use, whereas the new OIDs for fans actually give the RPM in an INTEGER format.

Old was a STRING format that just said "main" or "backup" in regards to which fan stage was actually being used. Useful as well but not as useful as the RPM ratings. I believe other devices also did the same as well that have supported this.

Comment (laf) [inline includes/discovery/sensors/fanspeeds/mikrotik.inc.php]: Strings won't have been discovered so we should be good now.
